src/shared/fstab-util.c

```
@@ -99,8 +99,17 @@ static int fstab_is_same_node(const char *what_fstab, const char *path) {
         if (path_equal(node, path))
                 return true;
 
-        if (is_device_path(path) && is_device_path(node))
+        if (is_device_path(path) && is_device_path(node)) {
+                /* If the original path to compare with is unavailable, return recognizable error. It's very likely
+                 * that udev hasn't created any devnode symlinks just yet, i.e. we cannot do reasonable comparison.
+                 *
+                 * NB: this is specifically not about the paths listed in fstab, because they OTOH can
+                 * realistically be or sometimes even expected to be missing, think: nofail mounts. */
+                if (access(path, F_OK) < 0)
+                        return -ENOTRECOVERABLE;
```

hmm, why the explicit access(F_OK) check here? devnode_same() does stat() internally anyway, which is kinda the same?

why ignore error codes here? why not check explicitly for ENOENT, because that's the error you are expecting here, no?